### ISSUE TITLE
OMEdit uses short class name, leading to circular reference and invalid model

### ISSUE THREAD

**Issue by mwetter** [OpenModelica/OpenModelica#13153]
### Description

When dragging a block into OMEdit, it uses the short class name. If the short class name is the same as the name of the class in which it is dropped, there is a circular reference, causing the model to be invalid.

### Steps to Reproduce

1. In OMEdit, open `Buildings.Controls.OBC.CDL.Discrete.Validation.FirstOrderHold`.
2. Drag in `Buildings.Controls.OBC.CDL.Discrete.FirstOrderHold`. (Note that the short class name of the instantiated block is the same as the one of the model in which it is instantiated.)
3. Try to set the `samplePeriod` of the newly instantiated block. The GUI shows the parameter entry, but entering a value gives an error.

The generated code is
```modelica
within Buildings.Controls.OBC.CDL.Discrete.Validation;
model FirstOrderHold
... [omitted] 
  FirstOrderHold firOrdHol1(samplePeriod = 0.2)  annotation(
    Placement(transformation(origin = {42, -30}, extent = {{-10, -10}, {10, 10}})));
equation
... [omitted]
```
The diagram is
![image](https://github.com/user-attachments/assets/5644be3f-d0ef-4c4a-b600-1145cc58cac0)

OMEdit issues the errors:
```
[6] 14:56:47 Translation Error
[Buildings.Controls.OBC.CDL.Discrete.Validation.FirstOrderHold: 17:29-17:47]: [Modified element samplePeriod not found in class FirstOrderHold.](omeditmessagesbrowser:///Buildings.Controls.OBC.CDL.Discrete.Validation.FirstOrderHold?lineNumber=17)

[7] 14:56:47 Translation Error
[Buildings.Controls.OBC.CDL.Discrete.Validation.FirstOrderHold: 17:3-18:84]: [Declaration of element firOrdHol1 causes recursive definition of class FirstOrderHold.](omeditmessagesbrowser:///Buildings.Controls.OBC.CDL.Discrete.Validation.FirstOrderHold?lineNumber=17)
```

### Expected Behavior

OMEdit should add the full class name for proper look-up.

### Version and OS

 - OpenModelica Version: OpenModelica 1.25.0~dev-143-g6a0e852
 - OS: Linux Ubuntu 22.04
 - Buildings, master, commit e775985e55e2c35f49f84b7d34e8b0951172dec4 (any recent release should behave the same in that regard)

**Comment by adeas31** [comment]
`addComponent` API makes the path relative. I guess a simple fix is to check if paths length are same.

**Comment by adeas31** [comment]
PR #13179 fixes the OMEdit.

**Comment by perost** [comment]
> PR #13179 fixes the OMEdit.

Do we still want to change `addComponent`? I already fixed it, but haven't made a PR yet.

**Comment by adeas31** [comment]
> > PR #13179 fixes the OMEdit.
> 
> Do we still want to change `addComponent`? I already fixed it, but haven't made a PR yet.

Yes. I only fixed the OMEdit part.

**Comment by perost** [comment]
> > > PR #13179 fixes the OMEdit.
> > 
> > 
> > Do we still want to change `addComponent`? I already fixed it, but haven't made a PR yet.
> 
> Yes. I only fixed the OMEdit part.

Ok, fixed in #13190.

**Comment by mwetter** [comment]
I confirm that this now works.
